ecr-login/api/factory.go

```diff
@@ -27,17 +29,26 @@ type Options struct {
 	CacheDir string
 }
 
+// ClientFactory is a factory for creating clients to interact with ECR
 type ClientFactory interface {
 	NewClient(awsSession *session.Session, awsConfig *aws.Config) Client
 	NewClientWithOptions(opts Options) Client
 	NewClientFromRegion(region string) Client
 	NewClientWithDefaults() Client
 }
+
+// DefaultClientFactory is a default implementation of the ClientFactory
 type DefaultClientFactory struct{}
 
+var userAgentHandler = request.NamedHandler{
+	Name: "ecr-login.UserAgentHandler",
+	Fn:   request.MakeAddToUserAgentHandler("amazon-ecr-credential-helper", version.Version),
```

Comment:
Do we need to change here to be consistent with User Agent syntax?

Reply:
Based on how `request.MakeAddToUserAgentHandler` works, I would expect it to result in the following user-agent:

aws-sdk-go/1.12.72 (go1.10; linux; amd64) amazon-ecr-credential-helper/0.0.1

I think this is fine.